src/anomalib/engine/engine.py

```diff
@@ -476,7 +480,11 @@ def train(
         self._setup_trainer(model)
         self._setup_dataset_task(train_dataloaders, val_dataloaders, test_dataloaders, datamodule)
         self.trainer.fit(model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
-        self.trainer.test(model, test_dataloaders, ckpt_path=ckpt_path, datamodule=datamodule)
+
+        if datamodule is not None and datamodule.test_split_mode == TestSplitMode.NONE:
```

Comment:
Would there be an edge case where test split is set to None, but test set is not empty? @djdameln ?
Maybe we could also check the length of the test set to be on the safe side?

Reply:
Do you mean something like this? If someone intentionally want to skip the test stage by setting the test split mode to None, but the test set is not empty, we give them a warning.

```python
if datamodule is not None and datamodule.test_split_mode == TestSplitMode.NONE:
    logger.info(f"The test_split_mode is set to '{TestSplitMode.NONE}'. Skipping test stage.")
    if len(datamodule.test_data) > 0:
        logger.warning(f"Found {len(datamodule.test_data)} images in the test set.")
    return None
```